ec2_snapshot: Update snapshot time format time

biohazd opened the pull request:
ansible ec2_snapshot ValueError: time data '2019-05-27T12:08:49.929Z' does not match format '%Y-%m-%dT%H:%M:%S.000Z'

##### SUMMARY
Fix error
ansible ec2_snapshot ValueError: time data '2019-05-27T12:08:49.929Z' does not match format '%Y-%m-%dT%H:%M:%S.000Z'

##### ISSUE TYPE
- Bugfix Pull Request

https://github.com/ansible/ansible/issues/57026

willthames commented on lib/ansible/modules/cloud/amazon/ec2_snapshot.py:
```diff
@@ -126,7 +126,7 @@
 # Find the most recent snapshot
 def _get_snapshot_starttime(snap):
-    return datetime.datetime.strptime(snap.start_time, '%Y-%m-%dT%H:%M:%S.000Z')
+     return datetime.datetime.strptime(snap.start_time, '%Y-%m-%dT%H:%M:%S.%fZ')
```
please remove the extra space at the start of the line to fix the failing tests

Akasurde commented:
@biohazd Thanks for the contribution.

biohazd commented:
it was all @glynastill

biohazd committed:
Update ec2_snapshot.py

ansible ec2_snapshot ValueError: time data '2019-05-27T12:08:49.929Z' does not match format '%Y-%m-%dT%H:%M:%S.000Z'

willthames commented:
I fixed up the whitespace. 

rebuild_merge

willthames commented:
Merged, thanks @biohazd and @glynastill

glynastill commented:
Many thanks @willthames
